src/BigBox.java

```diff
@@ -0,0 +1,18 @@
+import java.util.Date;
+
+public class BigBox extends Box {
+    private static final double BIG_BOX_FLAT_PRICE = 7.99;
+    private static final double BIG_BOX_PRICE_PER_MILLIGRAM = 0.99;
+
+    public BigBox(Type type, String address, Date date, double weight) {
```

Comment:
You could set the Type to BigBox here, directly, instead of letting the user potentially set a wrong one